isb_lib/sitemaps/sitemap_fetcher.py

```diff
@@ -94,7 +87,7 @@ def fetch_thing(self) -> ThingFetcher:
             # self.thing = None
             return self
 
-    def thing_identifier(self) -> str:
+    def thing_identifier(self) -> Optional[str]:
```

Comment:
Saying something was a `str` and returning `None` was a common error I had made.  Changing to `Optional` is the right fix as we legit return `None` in error cases, so the caller should check it.

Reply:
I've also seen `typing.Union[str, None]`, (I believe Optional is shorthand for Union) and [python 3.10 now](https://peps.python.org/pep-0604/) supports `v: str | None` which is nice if everything is on the version